src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,67 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { getMovie } from '../../api';
 import './FindMovie.scss';
+import { MovieCard } from '../MovieCard';
+import { MovieData } from '../../types/MovieData';
+import { movieDataToMovie } from '../../utils/movieDatatoMovie';
+import { ResponseError } from '../../types/ReponseError';
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  setMovies: React.Dispatch<React.SetStateAction<Movie[]>>;
+  hasMovie: (imdbId: string) => boolean;
+};
+
+export const FindMovie: React.FC<Props> = ({ setMovies, hasMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<MovieData | null>();
+  const [loading, setLoading] = useState(false);
+  const [isQueryAbsent, setIsQueryAbsent] = useState(true);
+
+  function isMovie(
+    object: MovieData | ResponseError,
+  ): object is MovieData {
+    return (object as MovieData).imdbID !== undefined;
+  }
+
+  const handleAddMovie = () => {
+    if (movie && !hasMovie(movie.imdbID)) {
+      setMovies(prev => [...prev, movieDataToMovie(movie)]);
+    }
+
+    setMovie(null);
+    setQuery('');
+  };
+
+  const handleQueryChange = (text: string) => {
+    setIsQueryAbsent(true);
+    setQuery(text);
+  };
+
+  const handleMovieRequest = (
```

Comment:
looks quite difficult to read. Do you have any ideas on how to improve its readability? Let me know if you got none, I'll help

Reply:
The parameter isQueryAbsent was starting as true and it was working the opposite way.
It's supposed to mean "is query not found". So I corrected that. 

Nothing else comes to my mind except writing a providePoster helper function to use in setMovie, instead of doing it right there. But I'd like to know if there is something else.